apps/builder/app/builder/features/style-panel/style-layers-list.tsx

```diff
@@ -76,47 +110,87 @@ export const LayersList = <
+    return hideLayer(property, index, value, createBatchUpdate);
   };
 
-  const onEditLayer = (index: number, newLayers: PropertyValueType) => {
-    return updateLayer<PropertyValueType>(
-      property,
-      newLayers,
-      layers,
-      index,
-      createBatchUpdate
-    );
+  const onEditLayer = (index: number, newLayers: TupleValue | LayersValue) => {
+    return updateLayer(property, newLayers, value, index, createBatchUpdate);
   };
 
   return (
     <CssValueListArrowFocus dragItemId={dragItemId}>
       <Flex direction="column" ref={sortableRefCallback}>
-        {layers.value.map((layer, index) => {
+        {value.value.map((layer, index) => {
           if (layer.type !== "tuple" && layer.type !== "function") {
             return null;
           }
           const id = String(index);
-          return renderLayer({
-            id,
-            index,
-            layer,
-            disabled,
-            property,
-            isHighlighted: dragItemId === id,
-            onLayerHide: handleHideLayer,
-            onDeleteLayer: handleDeleteLayer,
-            createBatchUpdate,
-            deleteProperty,
-            onEditLayer,
-          } as LayerProps<LayerType>);
+          const properties = extractPropertiesFromLayer(layer);
+          const { name, value, color } = properties;
+
+          return (
+            <FloatingPanel
+              key={index}
+              title={label}
+              content={renderContent({
+                index,
+                property,
+                layer,
+                onEditLayer,
+                propertyValue: value,
+                onDeleteLayer: handleDeleteLayer,
+                deleteProperty,
+              })}
+            >
+              <CssValueListItem
+                id={id}
+                draggable={true}
+                active={dragItemId === id}
+                index={index}
+                label={<Label truncate>{name}</Label>}
+                hidden={layer.type === "tuple" && layer?.hidden}
+                thumbnail={
+                  property === "textShadow" || property === "boxShadow" ? (
+                    <ColorThumb color={color} />
+                  ) : undefined
+                }
+                buttons={
+                  <>
+                    {layer.type === "tuple" ? (
+                      <SmallToggleButton
+                        variant="normal"
+                        pressed={layer?.hidden}
+                        disabled={false}
+                        tabIndex={-1}
+                        onPressedChange={() => handleHideLayer(index)}
+                        icon={
+                          layer?.hidden ? (
+                            <EyeconClosedIcon />
+                          ) : (
+                            <EyeconOpenIcon />
+                          )
+                        }
+                      />
+                    ) : null}
```

Comment:
for the sake of consistency
```suggestion
                    ) : undefined}
```